botocore/exceptions.py

```diff
@@ -100,7 +88,27 @@ class ServiceNotInRegionError(BotoCoreError):
     fmt = 'Service {service_name} not available in region {region_name}'
 
 
-class UnknownEndpointError(BotoCoreError):
+class BaseEndpointResolverError(BotoCoreError):
+    """Base error for endpoint resolving errors.
+
+    Should never be raised directly, but clients can catch
+    this exception if they want to generically handle any errors
+    during the endpoint resolution process.
+
+    """
+
+
+class NoRegionError(BaseEndpointResolverError):
+    """
+    No region was specified
+
+    :ivar env_var: The name of the environment variable to use to
+        specify the default region.
```

Comment:
We should probably remove the docs for `env_var` since it's no longer set or printed out.